src/utils/reqwest_resume/mod.rs

```diff
@@ -0,0 +1,165 @@
+//! Some modifications have been done to update the code regarding `tokio` and
+//! a change in dependency moving from `hyperx` to `hyper`.
+
+use bytes::Bytes;
+use futures::{ready, FutureExt, Stream, TryFutureExt};
+use hyper::header::{self, HeaderMap, HeaderValue};
+use std::{
+    future::Future,
+    pin::Pin,
+    task::{Context, Poll},
+    time::Duration,
+};
+use tokio::time::sleep;
+
+/// Extension to [`reqwest::Client`] that provides a method to convert it
+pub trait ClientExt {
+    /// Convert a [`reqwest::Client`] into a [`reqwest_resume::Client`](Client)
+    fn resumable(self) -> Client;
+}
+impl ClientExt for reqwest::Client {
+    fn resumable(self) -> Client {
+        Client(self)
+    }
+}
+
+/// A `Client` to make Requests with.
+///
+/// See [`reqwest::Client`].
+#[derive(Debug)]
+pub struct Client(reqwest::Client);
+impl Client {
+    /// Constructs a new `Client`.
+    ///
+    /// See [`reqwest::Client::new()`].
+    pub fn new() -> Self {
+        Self(reqwest::Client::new())
+    }
+    /// Convenience method to make a `GET` request to a URL.
+    ///
+    /// See [`reqwest::Client::get()`].
+    pub fn get(&self, url: reqwest::Url) -> RequestBuilder {
+        // <U: reqwest::IntoUrl>
+        RequestBuilder(self.0.clone(), reqwest::Method::GET, url)
+    }
+}
+
+/// A builder to construct the properties of a Request.
+///
+/// See [`reqwest::RequestBuilder`].
+#[derive(Debug)]
+pub struct RequestBuilder(reqwest::Client, reqwest::Method, reqwest::Url);
+impl RequestBuilder {
+    /// Constructs the Request and sends it the target URL, returning a Response.
+    ///
+    /// See [`reqwest::RequestBuilder::send()`].
+    pub fn send(&mut self) -> impl Future<Output = reqwest::Result<Response>> + Send {
+        let (client, method, url) = (self.0.clone(), self.1.clone(), self.2.clone());
+        async move {
+            let response = loop {
+                let builder = client.request(method.clone(), url.clone());
+                match builder.send().await {
+                    Err(err) if !err.is_builder() && !err.is_redirect() && !err.is_status() => {
+                        sleep(Duration::from_secs(1)).await
+                    }
+                    x => break x?,
+                }
+            };
+            let accept_byte_ranges =
+                if let Some(value) = response.headers().get(header::ACCEPT_RANGES) {
+                    value.as_bytes() == b"bytes"
+                } else {
+                    false
+                };
+            let resp = Response {
+                _client: client,
+                _method: method,
+                _url: url,
+                _response: response,
+                _accept_byte_ranges: accept_byte_ranges,
+                _pos: 0,
+            };
+            Ok(resp)
+        }
+    }
+}
+
+/// A Response to a submitted Request.
+///
+/// See [`reqwest::Response`].
+#[derive(Debug)]
+pub struct Response {
+    _client: reqwest::Client,
```

Comment:
Any reason for those fields to be `_`?

Reply:
This was to avoid lint errors (fields are never read). Maybe we could remove them all and return directly a `reqwest::Response`?

Reply:
They are never read, but the whole object is sent within a `Result`. I would consider doing `allow(unused)` or returning an actual `reqwest::Response` if that serves the purpose.

Reply:
The code was actually unused, so that explains why.